// ssh-sync/activate.js

/*
    require('ssh-sync')({
        local: {
            directory: <local directory>,
            exclude: [
                <path in local directory to exclude>,
                <path in local directory to exclude>,
                <path in local directory to exclude>
            ]
        },
        remote: {
            address: <ip address>,
            user: <username address>,
            key: <path to key path>,
            directory: <remote directory>
        }
    });
*/

var presets = {
    info: "verbose", // "percentage complete" (untested) or "verbose"
}

module.exports = function(macros) {
    const options = require("lodash/merge")({}, presets, macros);

    if (options.watch) {
        require('./lib/sync')(options)

        const watcher = require('chokidar').watch(`${ options.local.directory }/**/*`, {
            persistent: true,
            ignoreInitial: true
        }).on('all', (event, path) => {
            require('./lib/sync')(options)
        }); 
    }
    else {
        require('./lib/sync')(options)
    }
}
